scripts/generate_activity_script/src/script.ts

```diff
@@ -356,19 +396,607 @@ async function withdrawRandomNFT(
     const { code, hash } = await neutronWallet.clientCW.getTx(res.txHash);
     if (code !== 0) {
       return {
-        mode: MODE.WITHDRAW,
+        mode: NeutronAction.WITHDRAW,
+        txHash: hash,
+        reason: "Check up given hash",
+      };
+    }
+    return res;
+  } catch (e) {
+    return {
+      mode: NeutronAction.WITHDRAW,
+      txHash: null,
+      reason: e.message,
+    };
+  }
+}
+
+/* Function dedicated to do IBC transfer
+ * This function used by two other functions in this script:
+ *  - IBCToTransfer
+ *  - IBCFromTransfer
+ * We've to use signAndBroadcast instead of sendIbcTokens since second is deprecated
+ */
+async function IBCTransfer(
+  clientSG: SigningStargateClient,
+  addressFrom: string,
+  addressTo: string,
+  channel: string,
+  port: string,
+  amount: Coin,
+  fee: Coin
+): Promise<string> {
+  const { transactionHash } = await clientSG.signAndBroadcast(
+    addressFrom,
+    [
+      {
+        typeUrl: "/ibc.applications.transfer.v1.MsgTransfer",
+        value: {
+          sourcePort: port,
+          sourceChannel: channel,
+          token: amount,
+          sender: addressFrom,
+          receiver: addressTo,
+          timeoutHeight: "0",
+          timeoutTimestamp: String(
+            Math.floor(Date.now() / 1000) * 1e9 + 10 * 60 * 1e9
```

Comment:
Are you sure that this calculation are right?

Reply:
Yes. Since we can't have `timeoutHeight` and `timeoutTimestamp` at the same time to be "0", I used `timeoutTimestamp` because it's easier to calculate

`timeoutTimestamp` value measured in nanoseconds (1e9). Date time is in milliseconds (1e3)

Reply:
make sence